mongo/mongo_test.go

```diff
@@ -23,18 +23,42 @@ var (
 	URI        = "localhost:27017"
 )
 
+const (
+	returnContextKey = "want_return"
+	early            = "early"
+)
+
 // Mongo represents a simplistic MongoDB configuration.
 type Mongo struct {
 	Collection string
 	Database   string
 	URI        string
 }
 
+type TestModel struct {
+	State           string              `bson:"state"`
+	NewKey          int                 `bson:"new_key,omitempty"`
+	LastUpdated     time.Time           `bson:"last_updated"`
+	UniqueTimestamp bson.MongoTimestamp `bson:"unique_timestamp,omitempty"`
+}
+
+type Times struct {
+	LastUpdated     time.Time           `bson:"last_updated"`
+	UniqueTimestamp bson.MongoTimestamp `bson:"unique_timestamp,omitempty"`
```

Comment:
This functionality is not needed, we only need a unique timestamp at the document level, not in nested objects.

Reply:
that struct is only for the test